spec/acceptance/dataset_items_spec.rb

```diff
@@ -0,0 +1,481 @@
+                                 response_body_content: "\"start_time_seconds\":",
+                                 invalid_content: "end_time_seconds"
+                             })
+  end
+
+  post '/dataset_items/filter' do
+    let(:authentication_token) { reader_token }
+    let(:raw_post) { {
+      'filter' => {
+          'start_time_seconds' => {
+              'in' => ['7', '100', '4']
+          }
+      },
+      'projection' => {
+          'include' => ['id', 'start_time_seconds', 'audio_recording_id', 'creator_id']
+      }
+    }.to_json }
+    standard_request_options(:post, 'FILTER (as reader)', :ok,
+                             {
+                                 expected_json_path: 'data',
+                                 data_item_count: 0
+                             })
+  end
+
+  # sort by virtual column
+  post '/dataset_items/filter' do
+    let(:authentication_token) { reader_token }
+    let(:raw_post) {
+      {
+        'filter' => {
+            'start_time_seconds' => {
+                'in' => ['1', '3', '8']
+            }
+        },
+        'projection' => {
+            'include' => ['id', 'start_time_seconds', 'audio_recording_id', 'creator_id', 'order']
+        },
+        sorting: {
+            order_by: :priority,
+            direction: :asc
+        }
+      }.to_json
+    }
+
+    let!(:new_dataset_item) {
+
+      # is included in the filter
+      FactoryGirl.create(:dataset_item,
+                         creator: admin_user,
+                         dataset: dataset,
+                         audio_recording: audio_recording,
+                         start_time_seconds: 3,
+                         end_time_seconds: 4,
+                         order: 1).save!
+
+      # is included in the filter by start time
+      FactoryGirl.create(:dataset_item,
+                         creator: admin_user,
+                         dataset: dataset,
+                         audio_recording: audio_recording,
+                         start_time_seconds: 1,
+                         end_time_seconds: 2,
+                         order: 5).save!
+
+      # is not included in the filter by start time
+      FactoryGirl.create(:dataset_item,
+                         creator: admin_user,
+                         dataset: dataset,
+                         audio_recording: audio_recording,
+                         start_time_seconds: 5,
+                         end_time_seconds: 6,
+                         order: 2).save!
+
+      # is included in the filter
+      FactoryGirl.create(:dataset_item,
+                         creator: admin_user,
+                         dataset: dataset,
+                         audio_recording: audio_recording,
+                         start_time_seconds: 8,
+                         end_time_seconds: 80,
+                         order: 4).save!
+
+    }
+
+    example 'FILTER by start time and sort by virtual column', document: true do
+      do_request
```

Comment:
Nice!

- Add this is its own section (just a comment block)
- Use our standard testing mechanism
  - You might need to extend `acceptance_spec_helper`  in a similar fashion to https://github.com/QutEcoacoustics/baw-server/blob/ef0155e7a74ba0d7c5716a3793af3323eddb2cfe/spec/helpers/acceptance_spec_helper.rb#L331

Reply:
Fixed. Added a new option to the opts that are passed to standard_request_options called "order", where the user specifies the property and the expected values in order.